internal/osbuild2/zipl_inst_stage.go

```diff
@@ -13,20 +13,17 @@ type ZiplInstStageOptions struct {
 
 func (ZiplInstStageOptions) isStageOptions() {}
 
-type ZiplInstStageDevices map[string]Device
-
-func (ZiplInstStageDevices) isStageDevices() {}
-
-type ZiplInstStageMounts []Mount
-
-func (ZiplInstStageMounts) isStageMounts() {}
-
-// Return a new zipl.inst stage. A device needs to be specified as 'disk' and root mountpoint must be provided
-func NewZiplInstStage(options *ZiplInstStageOptions, devices *CopyStageDevices, mounts *CopyStageMounts) *Stage {
+// Return a new zipl.inst stage. The 'disk' parameter must represent the
+// (entire) device that contains the /boot partition.
+func NewZiplInstStage(options *ZiplInstStageOptions, disk *Device, devices *Devices, mounts *Mounts) *Stage {
+	// create a new devices map and add the disk to it
+	devmap := map[string]Device(*devices)
+	devmap["disk"] = *disk
```

Comment:
Shouldn't this be defined as a separate Device(s) type with property `disk`, instead of `device`?

Reply:
That looks fine to me. In the manifest, we need:

```
"devices": {
   "disk": {
     <a device, probably loopback>
   }
}
```
Maybe I am missing something though

Reply:
Yeah, I should have resolved this comment. It was made before our session with Achilleas to sort out how to actually represent devices and various input types in composer. I didn't realise that the key under which the input device is expected is stage-specific...